generator/internal/descriptor_utils.cc

```diff
@@ -238,29 +237,23 @@ void SetResourceRoutingMethodVars(
 
 std::string DefaultIdempotencyFromHttpOperation(
     google::protobuf::MethodDescriptor const& method) {
-  if (!method.options().HasExtension(google::api::http)) {
-    // if no http option exists, opt for more conservative.
-    return "kNonIdempotent";
-  }
-
-  google::api::HttpRule http_rule =
-      method.options().GetExtension(google::api::http);
-
-  switch (http_rule.pattern_case()) {
-    case google::api::HttpRule::kGet:
-    case google::api::HttpRule::kPut:
-      return "kIdempotent";
-      break;
-    case google::api::HttpRule::kPost:
-    case google::api::HttpRule::kDelete:
-    case google::api::HttpRule::kPatch:
-      return "kNonIdempotent";
-      break;
-    default:
-      GCP_LOG(FATAL) << __FILE__ << ":" << __LINE__
-                     << ": google::api::HttpRule not handled" << std::endl;
-      std::exit(1);
+  if (method.options().HasExtension(google::api::http)) {
```

Comment:
FYI, this file seems to have several additional cleanups, unrelated to the main change. Fine by me, but a bit harder to grok.

Reply:
I did not intend to change any behavior here ... just invert some tests so that `GCP_LOG(FATAL)` is not the last thing in a non-void function.  This is the "[[unreachable]] refactoring" I mentioned in the description, although undoubtedly poorly expressed.  It would be nice if the compiler could clearly see that a FATAL log does not return.

Reply:
Ah, SGTM then.